rs/sns/testing/tests/sns_testing_ci.rs

```diff
@@ -62,3 +105,143 @@ async fn test_sns_testing_pocket_ic() {
         format!("{}, {}!", new_greeting, test_call_arg),
     );
 }
+
+#[tokio::test]
+pub async fn test_missing_nns_canisters() {
+    let pocket_ic = PocketIcBuilder::new()
+        .with_nns_subnet()
+        .with_sns_subnet()
+        .with_ii_subnet()
+        .with_application_subnet()
+        .build_async()
+        .await;
+    bootstrap_nns(&pocket_ic, vec![], vec![], vec![]).await;
+    pocket_ic
+        .stop_canister(
+            LEDGER_INDEX_CANISTER_ID.get().into(),
+            Some(ROOT_CANISTER_ID.get().into()),
+        )
+        .await
+        .unwrap();
+    pocket_ic
+        .delete_canister(
+            LEDGER_INDEX_CANISTER_ID.get().into(),
+            Some(ROOT_CANISTER_ID.get().into()),
+        )
+        .await
+        .unwrap();
+    assert_eq!(
+        validate_network(&pocket_ic).await,
+        vec![SnsTestingNetworkValidationError::MissingNnsCanister(
+            "ledger-index".to_string(),
+        )]
+    )
+}
+
+#[tokio::test]
+pub async fn test_missing_sns_wasm_upgrade_steps() {
+    let pocket_ic = PocketIcBuilder::new()
+        .with_nns_subnet()
+        .with_sns_subnet()
+        .with_ii_subnet()
+        .with_application_subnet()
+        .build_async()
+        .await;
+    let mut nns_installer = NnsInstaller::default();
```

Comment:
I suppose we should have followed a single builder patter style for `PocketIcBuilder` and `NnsInstaller`. 

No change required for this PR